File: YoonJunSik/week_5/예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        //부서 신청 금액 오름차순 정렬
        Arrays.sort(d);

        int count = 0;
        int sum = 0;

        //작은 금액부터 더해가며 갯수를 세기
        for(int money : d)
        {
            if(sum + money > budget) // 예산 초과의 경우
            {
                break;
            }
            sum += money;
            count++;
        }
        return count;
    }
}

File: LeeTaeJeong/week5/예산.py
def solution(d, budget):
    answer = 0
    
    # 1. 신청 금액이 적은 순서대로 정렬합니다. (오름차순)
    d.sort()
    
    # 2. 적은 금액부터 차례대로 예산에서 뺍니다.
    for amount in d:
        if budget >= amount:
            budget -= amount
            answer += 1
        else:
            # 예산이 부족하면 더 이상 지원할 수 없으므로 반복을 종료합니다.
            # (뒤에 있는 부서들은 현재 amount보다 더 큰 금액을 요구하므로 볼 필요가 없습니다)
            break
            
    return answer